### PR TITLE
Issue #5501: NULL layout paths (e.g., Default layout) cause problems in PHP 8.1

### LINKED ISSUE #5501: NULL layout paths (e.g., Default layout) cause problems in PHP 8.1
## Description of the bug

In PHP 8.1, if we pass a NULL path to various `path.inc` functions like `path_is_admin($path)` (and any function that ends up at `backdrop_match_path($path, $patterns)`, in PHP 8.1, any automated test that does this will fail with this error:

```
Exception Deprecated path.inc          315                                                                             
    preg_match(): Passing null to parameter #2 ($subject) of type string is deprecated
```

This showed up when working on PR https://github.com/backdrop/backdrop/pull/3955, and that contains a workaround. But there's a fundamental issue in core, which ultimately stems from the fact that for the default layout, `getPath()` will return NULL, rather than, say, an empty string.

## Steps To Reproduce

I've added a test in the PR for this issue that creates the problem. It contains the following:

```
    $layouts = layout_load_multiple_by_path('');
    $default_layout = $layouts['default'];
    $path = $default_layout->getPath(); // this will be NULL
    $this->assertTrue(is_null($path), 'Default layout path is NULL.');

    backdrop_static_reset('path_is_admin');
    $result = path_is_admin($path); // This should throw a deprecation warning in PHP 8.1
    $this->assertFalse($result, 'Default layout is not an admin path.');

    backdrop_match_path($path, "foo\nbar"); // This should throw a deprecation warning in PHP 8.1
```

## Actual behavior

The test will pass in PHP 5.3 and 7.4, but fails in PHP 8.1. 

## Expected behavior

Pass in all versions of PHP.

## Solution(s)

I could imagine two possible mitigations:

* Modify the default layout so that its path is an empty string, rather than NULL. (Not sure what other effects that might have.)
* Modify various functions in `path.inc` to gracefully handle NULL variables passed for the `$path` variable.

I solicit further input and thoughts on the best approach.

### PR DESCRIPTION
**Pull request by quicksketch**
Fixes https://github.com/backdrop/backdrop-issues/issues/5501.

Builds on (and may replace) https://github.com/backdrop/backdrop/pull/3958 which was tests only. This should pass on PHP 8.1 by casting the given `$path` value to a string to handle `NULL` values.

### CODE REVIEW

File: core/modules/layout/tests/layout.test [hunk 1]
```diff
@@ -3414,3 +3414,39 @@ class LayoutFlexibleTemplateTest extends BackdropWebTestCase {
...
+    $this->admin_user = $this->backdropCreateUser(array(
+      'access content',
+      'administer layouts',
+      'access user profiles',
+    ));
+    $this->backdropLogin($this->admin_user);
+  }
+
+  /**
+   * Test that the correct renderer is used
```
Comment (indigoxela) [inline core/modules/layout/tests/layout.test:3438]: I don't understand that description ... the test does not check the renderer. Or am I missing something?

File: core/modules/layout/tests/layout.test [hunk 2]
```diff
@@ -3414,3 +3414,39 @@ class LayoutFlexibleTemplateTest extends BackdropWebTestCase {
...
+    $layouts = layout_load_multiple_by_path('');
+    $default_layout = $layouts['default'];
+    $path = $default_layout->getPath(); // this will be NULL
+    $this->assertTrue(is_null($path), 'Default layout path is NULL.');
+
+    backdrop_static_reset('path_is_admin');
+    $result = path_is_admin($path); // This should throw a deprecation warning in PHP 8.1
+    $this->assertFalse($result, 'Default layout is not an admin path.');
+
+    backdrop_match_path($path, "foo\nbar"); // This should throw a deprecation warning in PHP 8.1
```
Comment (indigoxela) [inline core/modules/layout/tests/layout.test:3450]: Coding standards, that line's too long, and inline comments should end with proper punctuation. Maybe better move it above the code line.

I'm not sure what this test is actually supposed to do, BTW. Checking if PHP 8.1 throws a deprecation warning? Why testing that?

File: core/modules/layout/tests/layout.test [hunk 3]
```diff
@@ -3414,3 +3414,39 @@ class LayoutFlexibleTemplateTest extends BackdropWebTestCase {
...
+      'administer layouts',
+      'access user profiles',
+    ));
+    $this->backdropLogin($this->admin_user);
+  }
+
+  /**
+   * Test that the correct renderer is used
+   */
+  function testLayoutPath() {
```
Comment (indigoxela) [inline core/modules/layout/tests/layout.test:3440]: public function testLayoutPath() {...

File: core/modules/layout/tests/layout.test (hunk 1 above)
Comment (indigoxela) [reply core/modules/layout/tests/layout.test:3438]: Oh, and it should end with punctuation.

**Comment by bugfolder** [comment]
Closing in favor of https://github.com/backdrop/backdrop/pull/3958.

**Pull request closed without merging**